src/Symfony/Component/Translation/Dumper/FileDumper.php

```diff
@@ -22,68 +22,14 @@
  *
  * @author Michel Salib <michelsalib@hotmail.com>
  */
-abstract class FileDumper implements DumperInterface
+abstract class FileDumper extends AbstractFileDumper
```

Comment:
I didn't see why we need another class `AbstractFileDumper`, IMO the safe way is to add `formatCatalogue` in `FileDumper` and override it in `AbstractTreeFileDumper` :

```patch
-            // save file
-            file_put_contents($fullpath, $this->format($messages, $domain));
+
+            $this->formatCatalogue($messages, $domain, $options);
         }
     }
 
+    protected function formatCatalogue(MessageCatalogue $messages, $domain, $options = array())
+    {
+        // save file
+        file_put_contents($fullpath, $this->format($messages, $domain));
+    }
+

```